File: Visitor Pattern/1. Minimal/Java/ConcreteVisitorA.java
public class ConcreteVisitorA implements Visitor {
    @Override
    public void visit(ConcreteElementA element) {
        System.out.println("ConcreteVisitorA visits ConcreteElementA: " + element.operationA());
    }
  
    @Override
    public void visit(ConcreteElementB element) {
        System.out.println("ConcreteVisitorA visits ConcreteElementB: " + element.operationB());
    }
}

File: Visitor Pattern/1. Minimal/Python/concrete_visitor_a.py
from visitor import Visitor

class ConcreteVisitorA(Visitor):
    def visit_concrete_element_a(self, element_a):
        print("ConcreteVisitorA visits ConcreteElementA:", element_a.operation_a())

    def visit_concrete_element_b(self, element_b):
        print("ConcreteVisitorA visits ConcreteElementB:", element_b.operation_b())
